master/internal/api_workspace.go

```diff
@@ -551,11 +559,16 @@ func (a *apiServer) DeleteWorkspace(
 		return nil, err
 	}
 
+	models_exist, _ := a.workspaceHasModels(ctx, req.Id)
+	if models_exist {
+		return nil, fmt.Errorf("workspace (%d) contains models; move or delete models first", req.Id)
+	}
+
 	holder := &workspacev1.Workspace{}
-	err = a.m.db.QueryProto("deletable_workspace", holder, req.Id)
+	// TODO(kristine): update workspace state in transaction with template delete
+	a.m.db.QueryProto("deletable_workspace", holder, req.Id)
 	if holder.Id == 0 {
-		return nil, errors.Wrapf(err, "workspace (%d) does not exist or not deletable by this user",
-			req.Id)
+		return nil, fmt.Errorf("workspace (%d) does not exist or not deletable by this user", req.Id)
```

Comment:
Updating error formatting to use `fmt.Errorf` instead of `errors.Wrapf` in this function for two reasons. Mainly because `errors` is deprecated. Also, because the function was relying on `errors.Wrapf` returning `nil` if `err` was `nil`. This way, it is more explicit when the function returns a response and when it does not.